src/script/lua_api/l_noise.cpp

```diff
@@ -499,6 +511,49 @@ int LuaPcgRandom::l_rand_normal_dist(lua_State *L)
 	return 1;
 }
 
+int LuaPcgRandom::l_get_state(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	LuaPcgRandom *o = checkObject<LuaPcgRandom>(L, 1);
+
+	u64 state[2];
+	o->m_rnd.getState(state);
+
+	std::stringstream s_state;
+	
+	s_state << std::hex << std::setw(64) << std::setfill('0') << state[0];
+	s_state << std::hex << std::setw(64) << std::setfill('0') << state[1];
+	
+	lua_pushstring(L, s_state.str().c_str());
+	return 1;
+}
+
+int LuaPcgRandom::l_set_state(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	LuaPcgRandom *o = checkObject<LuaPcgRandom>(L, 1);
+	if (!lua_isstring(L, 2)) {
+		throw LuaError("PcgRandom.set_state(): Expected hex strings");
+	}
+
+	std::string l_string = lua_tostring(L, 2);
+	if (l_string.size()!=128) {
```

Comment:
Shouldn't this be 32 (if you update the above code to use widths of 16 for the two u64's)?